client/src/components/projects/project-filter/ProjectFilter.vue

```diff
@@ -0,0 +1,178 @@
+            <div class="filter-item">
+              <Checkbox :checked = "t.checked"
+                        v-on:update:checked="handleFilterAction(t, 'type')"/>
+              <span class="title is-5 is-size-16">{{ t.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          name="Domains"
+          :opened="domainAccordionOpened"
+          @update:opened="domainAccordionOpened=!domainAccordionOpened"
+        >
+          <div
+            v-for="domain in domains"
+            :key="domain.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "domain.checked"
+                        v-on:update:checked="handleFilterAction(domain, 'domain')"/>
+              <span class="title is-5 is-size-16">{{ domain.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          v-if="customers && customers.length"
+          name="Customers"
+          :opened="customerAccordionOpened"
+          @update:opened="customerAccordionOpened=!customerAccordionOpened"
+        >
+          <div
+            v-for="customer in convertedCustomers"
+            :key="customer.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "customer.checked"
+                        v-on:update:checked="handleFilterAction(customer, 'customer')" /> 
+              <span class="title is-5 is-size-16">{{ customer.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import {IProject} from '../../../shared/interfaces/project';
+  import Accordeon from '../../common/Accordeon/Accordeon.vue';
+  import Checkbox from '../../common/Checkbox/Checkbox.vue';
+  import Constants from '../../../shared/constants/constants';
+  import { Extension } from '../../../shared/classes/Extension';
+  
+
+  export default Vue.extend({
+      data() {
+        return {
+            // properties for accordion sections open/closed
+            lineAccordionOpened: true,
+            programAccordionOpened: true,
+            typeAccordionOpened: true,
+            domainAccordionOpened: true,
+            customerAccordionOpened: true,
+            technologyAccordionOpened: true,
+
+            // models for accordeon section
+            lines: new Array(),
+            domains: new Array(),
+            programs: new Array(),
+            types: new Array()
+        };
+      },
+
+      props: ['customers'],
+
+      computed: {
+        // customers come from parent
+        // no customers in constants.ts
+        convertedCustomers(): any[] {
+          return this.createModelForCheckboxes(this.customers);
+        }
+      },
+
+      components: {
+          Accordeon,
+          Checkbox
+      },
+      created() {
+         // convert for accordeon section with checkboxes
+         this.lines = this.createModelForCheckboxes(Constants['lines']);
```

Comment:
Should be a map - see above, with a type scpecified.
And here it would be just a forEach loop